bibtex-actions.el

```diff
@@ -122,13 +122,26 @@ The same string is used for display and for search."
     :group 'bibtex-actions
     :type  '(cons string string))
 
+(defcustom bibtex-actions-display-transform-functions
+  '((t  . bibtex-completion-clean-string)
```

Comment:
Why this?

Reply:
This is to remove things like braces and quoting from fields, it has been there from the start.

> I added a few minor suggestions and a question, but it may take me a bit of time to take a closer look, and to test.
> 
> Semester starts tomorrow.

Take your time, my pace will slow down too.

Reply:
Yes, I know, but doesn't parsebib do that?

I know a lot of BC code was done when parsebib was very new.

Reply:
For example, see `parsebib--parse-bib-value`

Reply:
> Yes, I know, but doesn't parsebib do that?
> 
> I know a lot of BC code was done when parsebib was very new.

Mostly but not always. It is probably not possible to always remove them without introducing ambiguity.

![2021-08-23T00:33:38,490186192+02:00](https://user-images.githubusercontent.com/88336724/130372168-6b58d3f7-bb60-48db-a1cb-1011977eba36.png)

Reply:
Here's the BC code:

```elisp
(defun bibtex-completion-clean-string (s)
  "Remove quoting and superfluous white space from BibTeX field value in S."
  (if s (replace-regexp-in-string "[\n\t ]+" " "
         (replace-regexp-in-string "[\"{}]+" "" s))
    nil))
```

Reply:
> Here's the BC code:
> 
> ```emacs-lisp
> (defun bibtex-completion-clean-string (s)
>   "Remove quoting and superfluous white space from BibTeX field value in S."
>   (if s (replace-regexp-in-string "[\n\t ]+" " "
>          (replace-regexp-in-string "[\"{}]+" "" s))
>     nil))
> ```

Do you want define the function ourselves in view of the planned removal of BC dependency? I can do that.

BTW, since I didn't mentioned it earlier, the screenshot is what I get if I remove  `(t  . bibtex-completion-clean-string)` from `bibtex-actions-display-transform-functions`.

Reply:
> Do you want define the function ourselves in view of the planned removal of BC dependency? I can do that.

I think so, so that we can just replace little pieces as we go?

Might be it becomes simpler too; IDK.

Reply:
> I think so, so that we can just replace little pieces as we go?
> 
> Might be it becomes simpler too; IDK.

Done!